theano/sandbox/cuda/blas.py

```diff
@@ -2506,11 +2502,11 @@ def c_code(self, node, nodename, inp, out, sub):
             || (CudaNdarray_HOST_DIMS(%(gx)s)[3] !=
                 CudaNdarray_HOST_DIMS(%(x)s)[3]))
         {
+            // init with zeros
+            // for st larger than ds some pixels in the input will be skipped
             Py_XDECREF(%(gx)s);
-            %(gx)s = (CudaNdarray*)CudaNdarray_New();
-            if ((NULL == %(gx)s)
-                || CudaNdarray_alloc_contiguous(%(gx)s, 4,
-                                                CudaNdarray_HOST_DIMS(%(x)s)))
+            %(gx)s = (CudaNdarray*)CudaNdarray_ZEROS(4, (int*)CudaNdarray_HOST_DIMS(%(x)s));
```

Comment:
Can you do the call to ZEROS only when it is needed? Otherwise, it could cause speed regression compared to the current code.